### ISSUE TITLE
Cannot read property 'charAt' of undefined

### ISSUE THREAD

**Issue by kerwinpeng** [npm/npm#7870]
```
2545 verbose addRemoteGit git fetch -a origin (git@github.com:lightning-viz/d3-tip.git)
2546 verbose addRemoteGit skipping chownr on Windows
2547 verbose addRemoteGit original treeish: master
2548 info git [ 'rev-list', '-n1', 'master' ]
2549 silly addRemoteGit resolved treeish: 2584232bdd3bf0b0a4fdc79917bd0d0848b1e0a2
2550 verbose stack TypeError: Cannot read property 'charAt' of undefined
2550 verbose stack     at getResolved (C:\nodejs\node_modules\npm\lib\cache\add-remote-git.js:357:13)
2550 verbose stack     at C:\nodejs\node_modules\npm\lib\cache\add-remote-git.js:220:23
2550 verbose stack     at C:\nodejs\node_modules\npm\lib\utils\git.js:26:8
2550 verbose stack     at ChildProcess.exithandler (child_process.js:742:7)
2550 verbose stack     at ChildProcess.emit (events.js:110:17)
2550 verbose stack     at maybeClose (child_process.js:1015:16)
2550 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
2551 verbose cwd D:\lightning\lightning
2552 error Windows_NT 6.1.7601
2553 error argv "C:\\nodejs\\\\node.exe" "C:\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
2554 error node v0.12.2
2555 error npm  v2.7.4
2556 error Cannot read property 'charAt' of undefined
2557 error If you need help, you may report this error at:
2557 error     <https://github.com/npm/npm/issues>
2558 verbose exit [ 1, true ]
```

**Comment by KenanY** [comment]
@kerwinpeng Could you try [updating](https://github.com/npm/npm/wiki/Troubleshooting#upgrading-on-windows) to either `npm@2.7.5` or `npm@2.7.6`?

**Comment by dstokes** [comment]
this is fixed in `2.8.4`

**Comment by othiym23** [comment]
`npm@2.7.6` includes the two commits that resolve #7829, which means that it should be the first version containing this fix. Either way, it's fixed now, so closing this as resolved.
